src/handlers/canonicalFieldsHandlers.ts

```diff
@@ -1,13 +1,37 @@
+import { getLogger } from '../logger';
+import { db } from '../database';
+import { isCanonicalFieldArray } from '../types';
+import { ValidationError } from '../errors';
 import type {
   Request,
   Response,
 } from 'express';
 
-function getCanonicalFields(req: Request, res: Response): void {
-  res.status(200)
-    .contentType('application/json')
-    .send('{ "hello": "world"}');
-}
+const logger = getLogger(__filename);
+
+const getCanonicalFields = (req: Request, res: Response): void => {
+  db.sql('canonicalFields.fetchAll')
+    .then((canonicalFields) => {
+      logger.debug(canonicalFields, 'canonicalFields.fetchAll');
+      const { rows } = canonicalFields;
+      if (isCanonicalFieldArray(rows)) {
+        res.status(200)
+          .contentType('application/json')
+          .send(rows);
+      } else {
+        throw new ValidationError(
+          'The database responded with an unexpected format.',
+          isCanonicalFieldArray.errors,
+        );
+      }
+    })
+    .catch((error: unknown) => {
+      logger.error(error);
```

Comment:
With the five levels provided in `slf4j` (the ones I'm most familiar with), `error` is the most severe, and so I would usually reserve that for situations where the process is either dying or about to die, and use `warn` in this kind of situation. But I think we have another more severe level `fatal` in `pino` so I guess we could use that one in that case. In other words: it's OK but I had a moment of nitpickiness that I thought I'd share.

Reply:
Word (appreciate the thinking / sharing).

Argh, I used to have a really good writeup of log levels but I can't find it any more.  The closest I could find was this documentation at [Winston](https://github.com/winstonjs/winston#logging-levels) which is in turn driven by this document: https://datatracker.ietf.org/doc/html/rfc5424

```
              0       Emergency: system is unusable
              1       Alert: action must be taken immediately
              2       Critical: critical conditions
              3       Error: error conditions
              4       Warning: warning conditions
              5       Notice: normal but significant condition
              6       Informational: informational messages
              7       Debug: debug-level messages
```

So basically I figured this is an "error condition" since it's something that actually did go wrong.

I think the difference between `fatal` and `error` is whether the system is able to recover from it.

That said... those are 8 levels and we don't have all 8 in pino, and `error` is peak of npm...

Dang it I really wanna find the writeup since this would be good to be consistent on so it's worth spending a moment on alignment.

Reply:
Ah this is the one!

https://stackoverflow.com/a/2031209/159522

```
    Trace - Only when I would be "tracing" the code and trying to find one part of a function specifically.
    Debug - Information that is diagnostically helpful to people more than just developers (IT, sysadmins, etc.).
    Info - Generally useful information to log (service start/stop, configuration assumptions, etc). Info I want to always have available but usually don't care about under normal circumstances. This is my out-of-the-box config level.
    Warn - Anything that can potentially cause application oddities, but for which I am automatically recovering. (Such as switching from a primary to backup server, retrying an operation, missing secondary data, etc.)
    Error - Any error which is fatal to the operation, but not the service or application (can't open a required file, missing data, etc.). These errors will force user (administrator, or direct user) intervention. These are usually reserved (in my apps) for incorrect connection strings, missing services, etc.
    Fatal - Any error that is forcing a shutdown of the service or application to prevent data loss (or further data loss). I reserve these only for the most heinous errors and situations where there is guaranteed to have been data corruption or loss.
```

I think if we subscribe to this approach then this would be fatal to the immediate operation (e.g. the fields are not returned so the GET operation has failed).

Reply:
(I'll mark this as resolved for now but we can pick this up in an issue for sure, and should probably plan to document the thinking)

Reply:
In general, I am OK with those descriptions. In particular, I most agree with the `info` and `warn` levels there: info is normal operation messages while `warn` is odd but recoverable. As for the descriptions of debug, trace, error, and fatal, there's wiggle room and a lot of subjectivity (and I am not sure what the poster means by "data loss" either) but again I think it's good overall, especially if we collapse the descriptions of trace/debug and error/fatal. In this situation, I think `error` sounds fine to me and if express is about to keel over and die we can use `fatal` in that situation.